portal/server/companies-api.test.js

```diff
@@ -0,0 +1,153 @@
+import axios, { AxiosError } from 'axios';
+import { MOCK_COMPANY_REGISTRATION_NUMBERS } from '@ukef/dtfs2-common';
+
+const { when } = require('jest-when');
+const { getCompanyByRegistrationNumber } = require('./companies-api');
+
+require('dotenv').config();
+
+const { PORTAL_API_URL } = process.env;
+
+const registrationNumber = MOCK_COMPANY_REGISTRATION_NUMBERS.VALID;
+const token = 'a token';
+
+const portalApiGetCompanyResponse = {
+  companiesHouseRegistrationNumber: MOCK_COMPANY_REGISTRATION_NUMBERS.VALID,
+  companyName: 'TEST COMPANY LTD',
+  registeredAddress: {
+    addressLine1: '1 Test Street',
+    locality: 'Test City',
+    postalCode: 'A1 2BC',
+    country: 'United Kingdom',
+  },
+  industries: [],
+};
+
+let axiosGetActual;
+let axiosGetMock;
+
+describe('getCompanyByRegistrationNumber()', () => {
+  beforeAll(() => {
+    axiosGetActual = axios.get;
+    axiosGetMock = jest.fn();
+    axios.get = axiosGetMock;
+  });
+
+  afterAll(() => {
+    axios.get = axiosGetActual;
+  });
+
+  it('returns the company if it is returned by the request to Portal API', async () => {
+    mockAxiosGetReturning({ status: 200, data: portalApiGetCompanyResponse });
+
+    const response = await getCompanyByRegistrationNumber(registrationNumber, token);
+
+    expect(response).toEqual({ company: portalApiGetCompanyResponse });
+  });
+
+  it('returns the correct error information if it is called without a registration number', async () => {
+    const response = await getCompanyByRegistrationNumber(undefined, token);
+
+    expect(response).toEqual({
+      errorMessage: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with a null registration number', async () => {
+    const response = await getCompanyByRegistrationNumber(null, token);
+
+    expect(response).toEqual({
+      errorMessage: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with a registration number that is the empty string', async () => {
+    const response = await getCompanyByRegistrationNumber('', token);
+
+    expect(response).toEqual({
+      errorMessage: 'Enter a Companies House registration number',
+    });
+  });
+
+  it('returns the correct error information if it is called with an invalid registration number', async () => {
+    const response = await getCompanyByRegistrationNumber(MOCK_COMPANY_REGISTRATION_NUMBERS.INVALID_TOO_SHORT, token);
+
+    expect(response).toEqual({
+      errorMessage: 'Enter a valid Companies House registration number',
+    });
+  });
+
+  it.each([
+    {
```

Comment:
* Variable
* Please add `500`  and I suppose `429` from APIM to `portal` will be `500`?